File: Phase08/SearchEngineEfCore/SearchEngineEfCore/FileReader.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text.RegularExpressions;

namespace SearchEngineEfCore
{
    public class FileReader : IFileReader
    {
        private static readonly Regex NonCharRegex = new Regex("[^a-zA-Z0-9 -]");
        private readonly string _folderPath;
        private readonly string _stopWordsPath;

        public FileReader(string folderPath, string stopWordPath)
        {
            _folderPath = folderPath;
            _stopWordsPath = stopWordPath;
        }

        public List<FileTuple> GetFilesContents()
        {
            return Directory.GetFiles(_folderPath)
                .Select(file => new FileTuple(Path.GetFileName(file), GetContentFromFile(file)))
                .ToList();
        }

        public HashSet<string> GetStopWords()
        {
            return new HashSet<string>(File.ReadAllLines(_stopWordsPath));
        }

        private string GetContentFromFile(string path)
        {
            return GetNormalizedString(File.ReadAllText(path));
        }

        private string GetNormalizedString(string content)
        {
            var replace = NonCharRegex.Replace(content, " ");
            return replace.ToLower();
        }
    }
}

File: Phase03/src/main/java/FileReader.java
package main.java;

import java.io.File;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;


import static java.lang.System.err;



public class FileReader implements IFileReader {
    private final File folder;
    private final String stopWordsPath;
    private static final String NON_CHAR_REGEX = "\\W+";

    public FileReader(String folderPath , String stopWord){
        folder = new File(folderPath);
        stopWordsPath = stopWord;
    }

    @Override
    public List<FileTuple> getFilesContents() {
        List<FileTuple> filesContent = new ArrayList<>();
        for (final File fileEntry : folder.listFiles()) {
            filesContent.add(new FileTuple(fileEntry.getName(),getContentFromFile(fileEntry)));
        }
        return filesContent;
    }

    @Override
    public List<String> getStopWords(){
        try {
            return Files.readAllLines(Paths.get(stopWordsPath));
        } catch (IOException e) {
            e.printStackTrace();
        }
        return new ArrayList<>();
    }



    @Override
    public String getContentFromFile(File doc) {
        String content = "";

        try {
            content = Files.readString(Path.of(doc.getPath()), StandardCharsets.UTF_8);
        } catch (IOException e) {
            err.println("error in read files");
            e.printStackTrace();
        }

        return getNormalizedString(content);
    }

    private String getNormalizedString(String content) {
        return content.toLowerCase().replaceAll(NON_CHAR_REGEX, " ");
    }

}
